src/Shared/EmptyListWrapper.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from 'react'
+import { View, Text, StyleSheet, Image } from 'react-native'
+import Icon from 'react-native-vector-icons/dist/MaterialIcons'
+
+import WrappedTextButton from '../TextButton/TextButton.js'
+
+export default class EmptyListWrapper extends Component {
+  render() {
+    let { listEmptyState, items, children } = this.props
+
+    if (!listEmptyState) {
+      return <>{children}</>
+    }
+
+    let {
+      buttonType,
+      buttonText,
+      buttonIcon,
+      buttonPrimaryColor,
+      buttonContrastColor,
+      buttonBorderRadius,
+      buttonShadow,
+      buttonUpperCase,
+      buttonAction,
+      buttonWidth,
+    } = listEmptyState
+
+    if (!items || items.length === 0) {
+      return (
+        <>
+          <ImageHolder listEmptyState={listEmptyState} />
+          {buttonType !== 'noButton' && (
+            <WrappedTextButton
+              type={buttonType}
+              text={buttonText}
+              icon={buttonIcon}
+              primaryColor={buttonPrimaryColor}
+              contrastColor={buttonContrastColor}
+              borderRadius={buttonBorderRadius}
+              shadow={buttonShadow}
+              upperCase={buttonUpperCase}
+              action={buttonAction}
+              container={{
+                width: buttonWidth,
+                alignSelf: 'center',
+              }}
+            />
+          )}
+        </>
+      )
+    } else {
+      return <>{children}</>
+    }
+  }
+}
+
+function ImageHolder(props) {
```

Comment:
```suggestion
function Content(props) {
```

I think a better name for this component would be `Content`, because it handles the logic for both images and Title/Subtitle...